Lib/gftools/stat.py

```diff
@@ -0,0 +1,350 @@
+        start, end = LINKED_VALUES[axis_tag]
+        for axis_value in axis["values"]:
+            if axis_value["value"] == start and end in seen_axis_values[axis_tag]:
+                axis_value["linkedValue"] = end
+    return stat_table
+
+
+def _add_elided_axis_values_to_stat(stat_table, elided_values):
+    """Overwrite which axis values should be elided.
+
+    Args:
+        stat: a stat table
+        elided_values: dict structured as {"axisTag": [100,200 ...]}"""
+    for axis_tag, axis in stat_table.items():
+        if axis_tag not in elided_values:
+            continue
+        for val in axis["values"]:
+            if val["value"] in elided_values[axis_tag]:
+                val["flags"] |= ELIDABLE_AXIS_VALUE_NAME
+            else:
+                val["flags"] &= ~ELIDABLE_AXIS_VALUE_NAME
+    return stat_table
+
+
+def _add_axis_value(style_name, value, flags=0x0, linked_value=None):
+    value = {"value": value, "name": style_name, "flags": flags}
+    if linked_value:
+        value["linkedValue"] = linked_value
+    return value
+
+
+def _default_axis_value(axis, axisreg=axis_registry):
+    axis_record = axisreg[axis]
+    default_value = axis_record.default_value
+    default_name = next(
+        (i.name for i in axis_record.fallback if i.value == default_value), None
+    )
+    return _add_axis_value(default_name, default_value, flags=ELIDABLE_AXIS_VALUE_NAME)
+
+
+def _validate_axis_order(axis_order, seen_axes):
+    axes_not_ordered = seen_axes - set(axis_order)
+    if axes_not_ordered:
+        raise ValueError(f"Axis order arg is missing {axes_not_ordered} axes.")
+
+
+def validate_family_fvar_tables(ttfonts):
+    """Google Fonts requires all VFs within a family to have the same
+    amount of axes and each axis range should match"""
+    for ttfont in ttfonts:
+        if "fvar" not in ttfont:
+            raise ValueError(f"Font is missing fvar table")
+
+    failed = False
+    src_fvar = ttfonts[0]['fvar']
+    src_axes = {a.axisTag: a.__dict__ for a in src_fvar.axes}
+    for ttfont in ttfonts:
+        fvar = ttfont['fvar']
+        axes = {a.axisTag: a.__dict__ for a in fvar.axes}
+        if len(axes) != len(src_axes):
+            failed = True
+            break
+        for axis_tag in axes:
+            if axes[axis_tag]['minValue'] != src_axes[axis_tag]['minValue']:
+                failed = True
+            if axes[axis_tag]['maxValue'] != src_axes[axis_tag]['maxValue']:
+                failed = True
+            # TODO should this fail if default values are different?
+    if failed:
+        raise ValueError("fvar axes are not consistent across the family")
+
+
+def _update_fvar_nametable_records(ttfont, stat_table):
+    nametable = ttfont["name"]
+    fvar = ttfont["fvar"]
+    family_name = font_familyname(ttfont)
+    axes_with_one_axis_value = [
+        a["values"][0] for a in stat_table if len(a["values"]) == 1
+    ]
+    tokens = [v["name"] for v in axes_with_one_axis_value]
```

Comment:
If there are no axes which contain only a single Axis Value, it means the font doesn't have any siblings. If this occurs, we should exit the function straight away.